internal/fs/hns_bucket_test.go

```diff
@@ -100,3 +100,101 @@ func (t *HNSBucketTests) TestDeleteFolder() {
 	_, err = os.Stat(dirPath)
 	assert.NotNil(t.T(), err)
 }
+
+func (t *HNSBucketTests) TestRenameFolderWithSrcDirectoryDoesNotExist() {
+	oldDirPath := path.Join(mntDir, "foo_not_exist")
+	newDirPath := path.Join(mntDir, "foo_rename")
+
+	err = os.Rename(oldDirPath, newDirPath)
+
+	assert.NotNil(t.T(), err)
+	_, err = os.Stat(newDirPath)
+	assert.NotNil(t.T(), err)
+}
+
+func (t *HNSBucketTests) TestRenameFolderWithDstDirectoryIsNotEmpty() {
+	oldDirPath := path.Join(mntDir, "foo")
+	_, err = os.Stat(oldDirPath)
+	assert.NoError(t.T(), err)
+	newDirPath := path.Join(mntDir, "bar")
+	_, err = os.Stat(newDirPath)
+	assert.NoError(t.T(), err)
+
+	err = os.Rename(oldDirPath, newDirPath)
+
+	assert.NotNil(t.T(), err)
+}
+
+func (t *HNSBucketTests) TestRenameFolderWithSameParent() {
+	oldDirPath := path.Join(mntDir, "foo")
+	_, err = os.Stat(oldDirPath)
+	assert.NoError(t.T(), err)
+	newDirPath := path.Join(mntDir, "foo_rename")
+
+	err = os.Rename(oldDirPath, newDirPath)
+
+	assert.NoError(t.T(), err)
+	_, err = os.Stat(oldDirPath)
+	assert.NotNil(t.T(), err)
```

Comment:
nit:

```suggestion
	assert.Error(t.T(), err)
```

Reply:
Also check for specific error code/string

Reply:
Generally it recommend to avoid comparing error strings.
https://bitfieldconsulting.com/posts/comparing-errors#:~:text=Comparing%20the%20string%20form%20of,should%20try%20to%20avoid%20it.&text=This%20is%20Part%202%20of,Comparing%20Go%20error%20values